aspnetcore/data/ef-rp/intro/samples/cu50/Pages/Students/Delete.cshtml.cs

```diff
@@ -64,13 +68,13 @@ public async Task<IActionResult> OnPostAsync(int? id)
                 await _context.SaveChangesAsync();
                 return RedirectToPage("./Index");
             }
-            catch (DbUpdateException /* ex */)
+            catch (DbUpdateException ex)
             {
-                //Log the error (uncomment ex variable name and write a log.)
+                _logger.LogError(1234, "Message:{msg} + Stack:{stack}", ex.Message, ex.StackTrace);
```

Comment:
@serpent5  is it OK to log stack trace without secure leaking? Note I added logging an a better message.
Rather than commented out logging, I think I will add logging to all the code snippets. It's practically free. Agree?

Reply:
> is it OK to log stack trace without leaking info?

Yeah, it's normal to log exceptions along with their stack-traces, etc, but I'd recommend going for the [`LogError` overload](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.logerror?view=dotnet-plat-ext-5.0#Microsoft_Extensions_Logging_LoggerExtensions_LogError_Microsoft_Extensions_Logging_ILogger_System_Exception_System_String_System_Object___) that takes the `Exception` instance:

```csharp
_logger.LogError(ex, "An error occurred when saving changes.");
```

This takes care of writing out the exception using a standard approach, which is also used by the framework when it picks up unhandled exceptions.

> I think I will add logging to all the code snippets

Yep, agreed. I expect that these were just comments because it's easier when building out a sample.